### PR TITLE
Extract common logic in HiveLocationService

### PR DESCRIPTION
**Pull request by shixuan-fan**

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveLocationService.java [hunk 1]
```diff
@@ -59,28 +58,25 @@ public LocationHandle forNewTable(SemiTransactionalHiveMetastore metastore, Conn
...
+    private LocationHandle createLocationHandle(HdfsContext context, ConnectorSession session, Path targetPath, boolean isExistingTable)
+    {
         if (shouldUseTemporaryDirectory(session, context, targetPath)) {
             Path writePath = createTemporaryPath(session, context, hdfsEnvironment, targetPath);
-            return new LocationHandle(targetPath, writePath, true, STAGE_AND_MOVE_TO_TARGET_DIRECTORY);
+            return new LocationHandle(targetPath, writePath, isExistingTable, STAGE_AND_MOVE_TO_TARGET_DIRECTORY);
         }
         else {
-            return new LocationHandle(targetPath, targetPath, true, DIRECT_TO_TARGET_EXISTING_DIRECTORY);
+            return new LocationHandle(targetPath, targetPath, isExistingTable, DIRECT_TO_TARGET_NEW_DIRECTORY);
```
Comment (wenleix) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveLocationService.java]: When will the write mode be `DIRECT_TO_TARGET_EXISTING_DIRECTORY` after the change?

Comment (shixuan-fan) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveLocationService.java]: Ah good catch!

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveLocationService.java [hunk 2]
```diff
@@ -59,28 +59,25 @@ public LocationHandle forNewTable(SemiTransactionalHiveMetastore metastore, Conn
...
+    }
 
+    private LocationHandle createLocationHandle(HdfsContext context, ConnectorSession session, Path targetPath, boolean isExistingTable)
+    {
         if (shouldUseTemporaryDirectory(session, context, targetPath)) {
             Path writePath = createTemporaryPath(session, context, hdfsEnvironment, targetPath);
-            return new LocationHandle(targetPath, writePath, true, STAGE_AND_MOVE_TO_TARGET_DIRECTORY);
+            return new LocationHandle(targetPath, writePath, isExistingTable, STAGE_AND_MOVE_TO_TARGET_DIRECTORY);
         }
         else {
```
Comment (wenleix) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveLocationService.java]: I would do
```java
else if (isExistingTable) {
    return new LocationHandle(targetPath, targetPath, true, DIRECT_TO_TARGET_EXISTING_DIRECTORY);
}
else {
    return new LocationHandle(targetPath, targetPath, false, DIRECT_TO_TARGET_NEW_DIRECTORY);
}
```

so there are 3 branches, and easier to follow (at least for myself :) )

**Comment by raghavsethi** [comment]
@shixuan-fan Could you follow commit message guidelines to start with a verb?

**Comment by shixuan-fan** [comment]
@raghavsethi Do you have any suggestions? I could say "Extract common logic in HiveLocationService". What do you think?

**Comment by raghavsethi** [comment]
@wxie that sounds good.

**Comment by shixuan-fan** [comment]
@raghavsethi I assume you are replying to me ;)

**Pull request merged**